vms/platformvm/validators/fee/fee.go

```diff
@@ -91,77 +91,46 @@ func (s State) CostOf(c Config, seconds uint64) uint64 {
 	return cost
 }
 
-// SecondsUntil calculates the number of seconds that it would take to charge at
-// least [targetCost] based on the dynamic fee mechanism. The result is capped
-// at [maxSeconds].
-func (s State) SecondsUntil(c Config, maxSeconds uint64, targetCost uint64) uint64 {
+// SecondsRemaining calculates the maximum number of seconds that a validator
+// can pay fees before their [costLimit] would be exceeded based on the dynamic
+// fee mechanism. The result is capped at [maxSeconds].
+func (s State) SecondsRemaining(c Config, maxSeconds uint64, costLimit uint64) uint64 {
 	// Because this function can divide by prices, we need to sanity check the
 	// parameters to avoid division by 0.
 	if c.MinPrice == 0 {
-		if targetCost == 0 {
-			return 0
-		}
 		return maxSeconds
```

Comment:
Would the correct fail-safe here be to return 0? It depends on how this will be used, but my guess is that zero will be more likely to result in a bailout while `maxSeconds` might result in unintended behaviour.

Reply:
To clarify a bit - this function is well defined for all input values. Returning `maxSeconds` here isn't intended as a flag that the input was invalid (if that was the case, this function would return an error). `maxSeconds` is the correct value to return in this case (as every second, the fee is equal to `0 nAVAX`. So, the validator can run forever).

Reply:
To clarify a bit, this function is essentially implementing:

"If you need to pay $x every second, how long does $y last?"

If you need to pay $0 every second, then it lasts forever.